src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -376,6 +376,10 @@ private void OnBatchScopeDisposed()
                             return s.RemoveAdditionalDocument(id);
                         });
 
+                    // Update project's order of documents.
+                    solution = _sourceFiles.UpdateProjectDocumentsOrderForBatch(solution, Id);
+                    solution = _additionalFiles.UpdateProjectDocumentsOrderForBatch(solution, Id);
```

Comment:
This won't work -- you'd need a helper to reorder additional documents too.

Reply:
Maybe the best thing to do here is remove "_additionalFiles" as we are not really that interested in ordering additional files.